src/mmw/js/src/modeling/views.js

```diff
@@ -98,15 +100,58 @@ var ScenariosView = Marionette.LayoutView.extend({
 
     addScenario: function() {
         var scenario = new models.ScenarioModel({
-                name: 'New Scenario'
-            });
+            name: this.makeNewScenarioName()
+        });
 
         this.collection.add(scenario);
     },
 
+    counter: 0,
+
+    scenarioCounter: function() {
+        this.counter++;
+        if (this.counter < this.collection.length) {
+            this.counter = this.collection.length;
+        }
+        return this.counter;
+    },
+
+    /**
+     * Return a new scenario name of the format "New Scenario X" where X is a
+     * positive number and that is greater than all previous X.
+     */
+    makeNewScenarioName: function() {
+        // When making new scenarios, we need to make sure we don't
+        // accidentally give two the same name. Use a counter but ensure slug
+        // name is not in use by looking at all the current names.
+        var numbers = _.without(_.map(this.collection.models, function(model) {
+            var slug = model.get('slug')
+            var regEx = /^new-scenario-(\d)+/g;
+            if (slug.match(regEx) !== null && slug.match(regEx).length === 1) {
+                return parseInt(model.get('slug').replace(/^new-scenario-/g, ''));
+            }
+        }), undefined);
+        if (!_.isEmpty(numbers)) {
+            var max = _.max(numbers) + 1;
+            return 'New Scenario ' + max;
+        } else {
+            return 'New Scenario ' + this.scenarioCounter();
+        }
+    },
+
     setActiveScenario: function(e) {
-        var scenarioSlug = $(e.target).data('scenario-slug');
+        var scenarioSlug = $(e.currentTarget).data('scenario-slug');
+
+        _.each(this.collection.models, function(model) {
```

Comment:
There is an `activeScenarioSlug` attribute on `project` (now `active-scenario-slug` in #230). I did it that way so we didn't have to worry about keeping that state on all the scenarios and we only have to update it in one place. If possible, consider using that attribute instead of `scenario.active`.